cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -277,33 +277,27 @@ public void scanApplications(ScanApplicationsRequest request,
     throws IOException {
 
     Range.Bound startBound = Range.Bound.INCLUSIVE;
-    Collection<Field<?>> startFields = request.getNamespaceId() == null ? Collections.emptyList() :
-      Collections.singletonList(Fields.stringField(
-        StoreDefinition.AppMetadataStore.NAMESPACE_FIELD,
-        request.getNamespaceId().getNamespace()));
     Range.Bound endBound = Range.Bound.INCLUSIVE;
+    Collection<Field<?>> startFields = Collections.emptyList();
+
+    if (request.getNamespaceId() != null) {
+      startFields = Collections.singletonList(Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD,
+                                                                 request.getNamespaceId().getNamespace()));
+    }
+    if (request.getAppReference() != null) {
```

Comment:
reorder ifs and use `if-else`. Currently for `AppReference` case both ifs will trigger. Also consider doing check for application as currently you create an object just to check if it's not null.

Reply:
Reordered ifs. 

Actually I created `ApplicationReference` obj to return also because I want to avoid exposing application name itself from the scanRequest. As we never want to scan applications without the namespace prefix key?

Reply:
I think it should be fine. Usually such things are inlined by JDK

Reply:
got it, I've updated it to return application. Thanks for the review!